### PR TITLE
libcoro: add version 0.7

### PR DESCRIPTION
**Pull request by chausner**
Specify library name and version:  **libcoro/0.7**

This adds a new recipe for [libcoro](https://github.com/jbaldwin/libcoro).

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by chausner** [comment]
> post_package(): WARN: [MISSING SYSTEM LIBS (KB-H043)] Library './lib/liblibcoro.so' links to system library 'm' but it is not in cpp_info.system_libs.

Not sure about this warning. The upstream `CMakeLists.txt` does not link to libm.

**Comment by SSE4** [comment]
> post_package(): WARN: [MISSING SYSTEM LIBS (KB-H043)] Library './lib/liblibcoro.so' links to system library 'm' but it is not in cpp_info.system_libs.

randomly noticed - is it really expected to have a double `lib`prefix? (`liblibcoro.so` -> `libcoro.so`?)

**Comment by chausner** [comment]
It's probably not intentional but this is the naming that the upstream project generates.
________________________________

**Review by jwillikers — APPROVED** [review]

File: recipes/libcoro/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("openssl/3.1.0", transitive_headers=True)
+        self.requires("c-ares/1.19.0", transitive_headers=True)
+        self.requires("tl-expected/1.1.0", transitive_headers=True)
```
Comment (prince-chrismc) [inline recipes/libcoro/all/conanfile.py:57]: https://github.com/jbaldwin/libcoro/blob/b815deb4b8eb44097c78af8a93ea5c4e47f4180e/inc/coro/semaphore.hpp#L3

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**